src/idvc/dvc_interface.py

```diff
@@ -3924,6 +3927,81 @@ def _addColorBar(self, viewer_widget):
         else:
             logging.warning('Wrong viewer type {}'.format(type(viewer)))
 
+    def _addColorBarRegistration(self, viewer):
+        # create the scalar_bar
+        scalar_bar = vtk.vtkScalarBarActor()
+        # scalar_bar.SetOrientationToHorizontal()
+        scalar_bar.SetOrientationToVertical()
+        
+        # Adjust the size of the scalar bar (set width and height as needed)
+        # Width as a fraction of the window or scene size
+        original_height = scalar_bar.GetHeight()
+        scalar_bar.SetHeight(0.4)  # Height as a fraction of the window or scene size
+        scalar_bar.SetWidth(scalar_bar.GetWidth() * scalar_bar.GetHeight() / original_height)
+        scalar_bar.SetTitle('Difference \nVref − Vcor')
+        fontsize = self.settings_window.fontsize_widget.value()
+        title_property = scalar_bar.GetTitleTextProperty()
+        title_property.SetJustificationToCentered()
+        title_property.SetFontSize(fontsize)
+        title_property.SetItalic(False)
+        print(scalar_bar.GetVerticalTitleSeparation())
+        scalar_bar.SetVerticalTitleSeparation(20)
+        label_property = scalar_bar.GetLabelTextProperty()
+        label_property.SetFontSize(fontsize)
+        label_property.SetItalic(False)
+        scalar_bar.SetTextPad(4)
+        #data = self.ref_image_data
+        #data = self.subtract.GetOutput()
+        #scalars = data.GetPointData().GetScalars()
+        #if scalars:
+        #    cmin, cmax = scalars.GetRange()
+        #else:
+        #    print("No scalars found in ref_image_data.")
+        #    cmin, cmax = 0, 1  # fallback
+        #print("cmin cmax", cmin, cmax)
+        lut2D = self._createLookupTableGrayscale()
+        #lut2D.SetTableRange(cmin, cmax)
+        #lut2D.Build()
+        scalar_bar.SetLookupTable(lut2D)
+        viewer.getRenderer().AddActor2D(scalar_bar)
+        print("I created the colorbar")
+        scalar_bar.SetVisibility(False)
+        self.scalar_bar = scalar_bar
+        
+
+    def _createLookupTableGrayscale(self):
+        lut = vtk.vtkLookupTable()
+        lut.SetNumberOfTableValues(256)
+        lut.SetRange(0.0, 1.0)  # Fixed range
+        
+        list(map(
+        lambda i: lut.SetTableValue(i, i / 255.0, i / 255.0, i / 255.0, 1.0),
+        range(256)
+    ))
+        lut.Build()
+        return lut
+
+    def updateColorBarFromDifferenceImage(self):
+        self.scalar_bar.SetVisibility(True)
+        # Get the output data from the subtract filter
+        data = self.subtract.GetOutput()
+
+        # Retrieve the scalar data from the output
+        scalars = data.GetPointData().GetScalars()
+
+        # Check if scalars exist in the data
+
+        # Get the min and max scalar values
+        cmin, cmax = scalars.GetRange()
```

Comment:
I would have thought that this would be sufficient.

```suggestion
        v = self.vis_widget_reg.frame.viewer
        cmin = v.ia.GetMinimum()
        cmax = v.ia.GetMaximum()
        
```

I don't see you connected any observer?